search-service/src/main/kotlin/com/egm/stellio/search/web/EntityHandler.kt

```diff
@@ -82,6 +83,72 @@ class EntityHandler(
         { it }
     )
 
+    /**
+     * Implements 6.5.3.4 - Merge Entity
+     */
+    @PatchMapping("/{entityId}", consumes = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
+    suspend fun merge(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @PathVariable entityId: String,
+        @RequestParam options: MultiValueMap<String, String>,
+        @RequestBody requestBody: Mono<String>
+    ): ResponseEntity<*> = either {
+        val entityUri = entityId.toUri()
+        val sub = getSubFromSecurityContext()
+
+        entityPayloadService.checkEntityExistence(entityUri).bind()
+
+        val body = requestBody.awaitFirst().deserializeAsMap()
+            .checkNamesAreNgsiLdSupported().bind()
+            .checkContentIsNgsiLdSupported().bind()
+        val observedAt = options.getFirst(QUERY_PARAM_OPTIONS_OBSERVEDAT_VALUE)
+            ?.parseTimeParameter("'observedAt' parameter is not a valid date")
+            ?.getOrElse { return@either BadRequestDataException(it).left().bind<ResponseEntity<*>>() }
```

Comment:
the `observedAt` parameter is optional

Reply:
the observedAt info is conveyed in a specific `observedAt` request parameter (see Table 6.5.3.4-1). a good opportunity to add a test.